docker/startup.sh

```diff
@@ -0,0 +1,75 @@
+# Entry script to allow docker to be ran for bash, tests and docs.
+# The script assumes a code repository can be mounted to ``/code`` and an output directory to ``/output``.
+# Executes ``mode`` on ``branch`` or the provided ``code`` directory.
+# $1: Mode, optional. Options:
+#        - test: execute unit tests
+#        - doc: build documentation, requires a mounted ``output`` directory if built from a branch.
+#        - if not provided: execute bash.
+# $2: Branch, optional.
+#        Mutually exclusive with mounting a ``code`` directory.
+#        Can be a branch on a Github fork, specified with the USERNAME#BRANCH format.
+#        The test or doc build is executed on this branch.
+
+if [ -z "$1" ]; then
+  echo "Executing in BASH mode."
+  bash
+  exit
+fi
+
+# doc and test modes require mounted directories and/or specified branches
+if ! [ -d "/code" ] && [ -z "$2" ]; then
+  echo "To perform $1 a code repository must be mounted to '/code' or a branch must be specified." >> /dev/stderr
+  exit 1
+fi
```

Comment:
why not elif?

Reply:
It doesn't really matter as `exit` is called if the body of any if-statement is entered.
I find this a bit more clear because you can reason about each statement individually without taking into account previous statements (it also indicates the order does not matter).

Reply:
Got it.